plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -259,76 +274,122 @@ def check_link_down_condition(self, port_obj, port_name, ports_counters):
-                dT = abs(port_obj.counters_values.get(Constants.TEMP_COUNTER, 0) - cable_temp)
-                port_obj.counters_values[Constants.TEMP_COUNTER] = cable_temp
-            if rcv_pkt_rate and error_rate / rcv_pkt_rate > self.max_pdr:
-                issues[port_name] = Issue(port_name, Constants.ISSUE_PDR)
-            elif self.temp_check and cable_temp and (cable_temp > self.tmax or dT > self.d_tmax):
-                issues[port_name] = Issue(port_name, Constants.ISSUE_OONOC)
-            elif self.no_down_count and link_downed_rate > 0:
-                link_downed_issue = self.check_link_down_condition(port_obj, port_name, ports_counters)
-                if link_downed_issue:
-                    issues[port_name] = link_downed_issue
-            if self.configured_ber_check:
-                symbol_ber_val = get_counter(Constants.PHY_SYMBOL_ERROR, row, default=None)
-                if symbol_ber_val is not None:
-                    ber_data = {
-                        Constants.TIMESTAMP : timestamp,
-                        Constants.SYMBOL_BER : symbol_ber_val, 
-                    }
-                    port_obj.ber_tele_data.loc[len(port_obj.ber_tele_data)] = ber_data
-                    port_obj.last_symbol_ber_timestamp = timestamp
-                    port_obj.last_symbol_ber_val = symbol_ber_val
-                    if not port_obj.active_speed or not port_obj.port_width:
-                        port_obj.active_speed, port_obj.port_width = self.get_port_metadata(port_name)
-                    if not port_obj.port_width:
-                        self.logger.debug(f"port width for port {port_name} is None, can't verify it's BER values")
-                        continue
-                    for (interval, threshold) in self.ber_intervals:
-                        symbol_ber_rate = self.calc_ber_rates(port_name, port_obj.active_speed, port_obj.port_width, interval)
-                        if symbol_ber_rate and symbol_ber_rate > threshold:
-                            issued_port = issues.get(port_name)
-                            if issued_port:
-                                issued_port.cause = Constants.ISSUE_PDR_BER
-                            else:
-                                issues[port_name] = Issue(port_name, Constants.ISSUE_BER)                    
-                            break                        
+            if pdr_issue:
+                issues[port_name] = pdr_issue
+            elif temp_issue:
+                issues[port_name] = temp_issue
+            elif link_downed_issue:
+                issues[port_name] = link_downed_issue
+            elif ber_issue:
+                issues[port_name] = ber_issue                        
         return issues
 
-    def calc_single_rate(self, port_name, port_speed, port_width, col_name, time_delta):
+    def calc_symbol_ber_rate(self, port_name, port_speed, port_width, col_name, time_delta):
```

Comment:
it would be great if you could write here a discription.

Reply:
done